src/pyqasm/visitor.py

```diff
@@ -1764,12 +1837,48 @@ def _visit_classical_assignment(
         # rvalue will be an evaluated value (scalar, list)
         # if rvalue is a list, we want a copy of it
         rvalue = statement.rvalue
+        lvar_base_type = lvar.base_type  # type: ignore[union-attr]
+        if isinstance(lvar_base_type, (qasm3_ast.DurationType, qasm3_ast.StretchType)):
```

Comment:
Similar comment as above, whenever we have validations , please refactor to a validator